crates/iceberg-catalog/src/service/storage/error.rs

```diff
@@ -45,6 +45,10 @@ impl From<TableConfigError> for ValidationError {
                     entity: "TableConfig".to_string(),
                 }
             }
+            TableConfigError::StsError(_, _) => ValidationError::Internal {
```

Comment:
I think this should not be internal, it happens on validation of a Warehouse. Its typically a misconfiguration on client side.

Reply:
it's 50/50, I'll split the error type then